Mage.Sets/src/mage/cards/b/BewitchingLeechcraft.java

```diff
@@ -0,0 +1,109 @@
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.ReplacementEffectImpl;
+import mage.abilities.effects.common.AttachEffect;
+import mage.abilities.effects.common.TapEnchantedEffect;
+import mage.abilities.effects.common.continuous.GainAbilityAttachedEffect;
+import mage.abilities.keyword.EnchantAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.counters.CounterType;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.target.TargetPermanent;
+import mage.target.common.TargetCreaturePermanent;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author Susucr
+ */
+public final class BewitchingLeechcraft extends CardImpl {
+
+    public BewitchingLeechcraft(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{1}{U}");
+        
+        this.subtype.add(SubType.AURA);
+
+        // Enchant creature
+        TargetPermanent auraTarget = new TargetCreaturePermanent();
+        this.getSpellAbility().addTarget(auraTarget);
+        this.getSpellAbility().addEffect(new AttachEffect(Outcome.BoostCreature));
+        this.addAbility(new EnchantAbility(auraTarget));
+
+        // When Bewitching Leechcraft enters the battlefield, tap enchanted creature.
+        this.addAbility(new EntersBattlefieldTriggeredAbility(new TapEnchantedEffect()));
+
+        // Enchanted creature has "If this creature would untap during your untap step,
+        // remove a +1/+1 counter from it instead. If you do, untap it."
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD,
+            new GainAbilityAttachedEffect(
+                new SimpleStaticAbility(new BewitchingLeechcraftReplacementEffect()),
+                AttachmentType.AURA
+            )
+        ));
+    }
+
+    private BewitchingLeechcraft(final BewitchingLeechcraft card) {
+        super(card);
+    }
+
+    @Override
+    public BewitchingLeechcraft copy() {
+        return new BewitchingLeechcraft(this);
+    }
+}
+
+class BewitchingLeechcraftReplacementEffect extends ReplacementEffectImpl {
+
+    public BewitchingLeechcraftReplacementEffect() {
+        super(Duration.EndOfGame, Outcome.Detriment);
+        staticText = "If this creature would untap during your untap step, " +
+            "remove a +1/+1 counter from it instead. If you do, untap it.";
+    }
+
+    public BewitchingLeechcraftReplacementEffect(final BewitchingLeechcraftReplacementEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public BewitchingLeechcraftReplacementEffect copy() {
+        return new BewitchingLeechcraftReplacementEffect(this);
+    }
+
+    @Override
+    public boolean checksEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.UNTAP
+            && game.getTurnStepType() == PhaseStep.UNTAP;
```

Comment:
I believe `checksEventType()` should only compare `event.getType()` and all other checks should be in `applies()` method.